runtime/src/snapshot_utils.rs

```diff
@@ -649,66 +901,188 @@ pub fn bank_from_snapshot_archive<P>(
 ) -> Result<(Bank, BankFromArchiveTimings)>
 where
     P: AsRef<Path> + std::marker::Sync,
+    Q: AsRef<Path> + std::marker::Sync,
 {
-    let unpack_dir = tempfile::Builder::new()
-        .prefix(TMP_SNAPSHOT_PREFIX)
-        .tempdir_in(snapshot_path)?;
-
-    let mut untar = Measure::start("snapshot untar");
-    let divisions = std::cmp::min(
+    let parallel_divisions = std::cmp::min(
         PARALLEL_UNTAR_READERS_DEFAULT,
         std::cmp::max(1, num_cpus::get() / 4),
     );
-    let unpacked_append_vec_map = untar_snapshot_in(
-        &snapshot_tar,
-        unpack_dir.as_ref(),
+
+    let full_snapshot_unarchive_preparation_result = do_snapshot_unarchive_preparation(
```

Comment:
nits: rename to `prepared_snapshot_archive`? a bit mouthful variable name, imo... like `Option<_>`, `Result<_>` type variable doesn't need to be suffixed with `_result`.

Reply:
I like it. Done!